library/std/src/sys/fs/unix.rs

```diff
@@ -262,7 +262,154 @@ impl ReadDir {
     }
 }
 
-struct Dir(*mut libc::DIR);
+pub struct Dir(*mut libc::DIR);
+
+// dirfd isn't supported everywhere
+#[cfg(not(any(
+    miri,
```

Comment:
we handle such missing platforms by returning errors instead of not having the methods exist at all